python/ray/train/gbdt_trainer.py

```diff
@@ -150,6 +154,7 @@ def __init__(
         label_column: str,
         params: Dict[str, Any],
         dmatrix_params: Optional[Dict[str, Dict[str, Any]]] = None,
+        num_boost_round: int = _DEFAULT_NUM_ITERATIONS,
```

Comment:
Should we make this a required argument instead?

Reply:
I think we should actually do the deprecation cycle (add a warning if user doesn't pass it in + make a tracker issue for 2.6). This would also solve the confusing config problem seen here https://anyscaleteam.slack.com/archives/C041HEHLANS/p1678846298426319

Reply:
But `lightgbm.train()`/`xgboost.train()` also default to 10 iterations, right? I think it's fine to just go with the default here. I think the main intent here is to explicitly pass the argument (and adjust on restore) so that it shows up in IDEs and can be tracked in the trainer?